src/herumi/signature.ts

```diff
@@ -41,44 +41,99 @@ export class Signature implements ISignature {
     return this.fromBytes(hexToBytes(hex));
   }
 
-  static aggregate(signatures: Signature[]): Signature {
+  static aggregate(signatures: SignatureArg[]): Signature {
     if (signatures.length === 0) {
       throw new EmptyAggregateError();
     }
 
     const context = getContext();
-    const signature = new context.Signature();
-    signature.aggregate(signatures.map((sig) => sig.value));
-    return new Signature(signature);
+    const agg = new context.Signature();
+    agg.aggregate(signatures.map(Signature.convertToSignatureType));
+    return new Signature(agg);
   }
 
-  static verifyMultipleSignatures(sets: {publicKey: PublicKey; message: Uint8Array; signature: Signature}[]): boolean {
+  static verifyMultipleSignatures(sets: SignatureSet[]): boolean {
+    if (!sets) throw Error("sets is null or undefined");
+
     const context = getContext();
     return context.multiVerify(
-      sets.map((s) => s.publicKey.value),
-      sets.map((s) => s.signature.value),
+      sets.map((s) => PublicKey.convertToPublicKeyType(s.publicKey)),
+      sets.map((s) => Signature.convertToSignatureType(s.signature)),
       sets.map((s) => s.message)
     );
   }
 
-  verify(publicKey: PublicKey, message: Uint8Array): boolean {
-    return publicKey.value.verify(this.value, message);
+  static async asyncVerifyMultipleSignatures(sets: SignatureSet[]): Promise<boolean> {
+    // eslint-disable-next-line no-console
+    console.log(
```

Comment:
do we really need to log this to console? I feel it's more suitable to put it in documentation instead

Reply:
I added a comment to the readme